src/pyrokinetics/gk_code/gs2.py

```diff
@@ -297,8 +297,28 @@ def get_local_species(self):
 
         return local_species
 
-    def _read_single_grid(self):
-        ky = self.data["kt_grids_single_parameters"]["aky"]
+    def _read_single_grid(self, local_geometry):
+
+        if (
+            "n0" in self.data["kt_grids_single_parameters"].keys()
```

Comment:
We only actually use `n0` if it is greater than zero so you might not want to rely just on the presence of the variable. We do also set a default `rhostar_single` so the user doesn't _have_ to set this, although they probably will.

Reply:
I've modified to use the default value of `n0` and `rhostar`